### PR TITLE
feat(project): resolve extends from `node_modules`

### PR DESCRIPTION
**Pull request by ematipico**
## Summary

This PR allows resolving a configuration file from `node_modules`. To do so, we use `oxc_resolver` to resolve the configuration file.

Unfortunately, I couldn't implement a test case using our memory file system. While `oxc_resolver` allows to implement a `FileSytem` trait to implement any possible logic, unfortunately, the `Resolver` wants ownership of type struct that implements `FileSystem`, e.g.:

```rs
struct MemoryFs {};
impl FileSystem for MemoryFs {};
let resolver = ResolverGeneric::with_file_system(options, MemoryFs);
```

This doesn't work well with our infrastructure, where our memory file system must outlive the entire application. 

I updated the documentation to explain how Biome will resolve the files, please read it and let me know if it's clear enough. Please leave any questions.

## Test Plan

I used our repository as a guinea pig. It means that the `check` command must pass.

### CODE REVIEW

File: crates/biome_diagnostics/src/adapters.rs [hunk 1]
```diff
@@ -106,3 +106,28 @@ impl Diagnostic for BpafError {
         Ok(())
     }
 }
+
+#[derive(Debug)]
+pub struct ResolveError {
+    error: oxc_resolver::ResolveError,
+}
```
[the remark this answers is not in the document]
Comment (ematipico) [reply crates/biome_diagnostics/src/adapters.rs:113]: All the adapters follow this pattern, and if we want to change it, we should do it for all of them. It is worth doing it in another PR

File: crates/biome_service/Cargo.toml [hunk 2]
```diff
@@ -44,6 +44,7 @@ dashmap                  = { workspace = true }
 ignore                   = { workspace = true }
 indexmap                 = { workspace = true, features = ["serde"] }
 lazy_static              = { workspace = true }
+oxc_resolver             = "1.4.0"
```
Comment (anonrig) [inline crates/biome_service/Cargo.toml:47]: Shouldn't this be workspace?

File: crates/biome_service/src/configuration/diagnostics.rs [hunk 3]
```diff
@@ -33,6 +34,8 @@ pub enum ConfigurationDiagnostic {
 
     /// Thrown when there's something wrong with the files specified inside `"extends"`
     CantLoadExtendFile(CantLoadExtendFile),
+
+    CantResolve(CantResolve),
```
Comment (anonrig) [inline crates/biome_service/src/configuration/diagnostics.rs:38]: Can you add a comment just like the rest of it?

File: shared/biome.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
+{
+	"$schema": "../packages/@biomejs/biome/configuration_schema.json",
+	"extends": ["shared/biome"],
```
Comment (Conaclos) [inline shared/biome.json:3]: This looks like a copy-paste issue.
```suggestion
```

File: shared/biome.json [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+			"recommended": true,
+			"style": {
+				"noNonNullAssertion": "off"
+			}
+		}
+	},
+	"organizeImports": {
+		"enabled": true
+	}
+}
```
Comment (Conaclos) [inline shared/biome.json:21]: What the purpose of this file? An example?

Comment (ematipico) [reply shared/biome.json:21]: It's written in the description:
> I used our repository as a guinea pig. It means that the check command must pass.

I am dog-fooding the feature using our repository

File: CHANGELOG.md [hunk 6]
```diff
@@ -56,6 +56,30 @@ Read our [guidelines for writing a good changelog entry](https://github.com/biom
 
 ### Configuration
 
+#### New features
+
+- Add the ability to resolve the configuration files defined inside `extends` from the `node_modules/` directory.
+
+  If you want to resolve a configuration file that matches the specifier `@org/configs/biome`, then your `package.json` file must look this:
```
Comment (anonrig) [inline CHANGELOG.md:63]: Can we add a link to Node.js resolver algorithm?

Comment (ematipico) [reply CHANGELOG.md:63]: Where and how? Feel free to add a suggestion. I don't know where that page is

Comment (ematipico) [reply CHANGELOG.md:63]: I added it in the documentation page

**Pull request merged**